### PR TITLE
api: getReaderSize - Supports Len() function

### PR DESCRIPTION
**Pull request by harshavardhana**
This is needed for certain packages which implement Len() instead
of Size() example *bytes.Buffer, *strings.Reader etc.

### CODE REVIEW

**Comment by krishnasrinivas** [comment]
had hit this requirement just a couple of hours ago when I was trying to PutObject a `bytes.NewBuffer([]byte) `stream

File: api-put-object.go [hunk 1]
```diff
@@ -27,78 +27,94 @@ import (
 	"strings"
 )
 
+// toInt - converts go value to its integer representation based
+// on the value kind if it is an integer.
+func toInt(value reflect.Value) (size int64) {
```
Comment (abperiasamy) [inline api-put-object.go]: You can directly do int() on reflect.ValueOf and let the compiler do the type validation.

**Comment by harshavardhana** [comment]
> You can directly do int() on reflect.ValueOf and let the compiler do the type validation.

That is not possible since reflect.Value is a proper type - this is what you get. 

https://play.golang.org/p/SeGQG1u8V-
```
prog.go:23: cannot convert result[0] (type reflect.Value) to type int
```

You have to know the Kind() to figure out and call the appropriate conversion function like value.Int().

**Comment by abperiasamy** [comment]
In that case why not use reflect.Value.Int()?
https://play.golang.org/p/bZPNIGLi8X

**Comment by harshavardhana** [comment]
> In that case why not use reflect.Value.Int()?
> https://play.golang.org/p/bZPNIGLi8X

In that case if the Value is not Int, it would panic.

**Pull request merged**